apple/swift-docc
# Rename `TitleStyle` to `PropertyListTitleStyle` to clarify its purpose

Bug/issue #, if applicable: rdar://126292460

## Summary

This renames the confusingly named `TitleStyle/title` and `TitleStyle/symbol` to `PropertyListTitleStyle/useDisplayName` and `PropertyListTitleStyle/useRawKey` to clarify what they are used for and fixes a fixes a call site that set the `titleStyle` of topic references because it wasn't clear that the API was only for property list keys. 

For the `TopicRenderReference` changes I started by renaming `titleStyle` to `propertyListTitleStyle`, `name` to `propertyListRawKey`, and `ideTitle` to `propertyListDisplayName`. Next, the common prefix led me to group these into a struct to make it even more clear that they are related.  

## Dependencies

None.

## Testing

In any documentation project
- Modify `/path/to/swift-docc-clone/bin/test-data-external-resolver` so that "kind/isSymbol" is "false"
- Add a documentation link to `<doc://com.test.bundle/something>` somewhere in the content
- Set `DOCC_LINK_RESOLVER_EXECUTABLE=/path/to/swift-docc-clone/bin/test-data-external-resolver`
- Preview documentation for the page with the link externally resolved link. 
   - The link text should not use a monospace font. 

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
